storage_service/locations/models/package.py

```diff
@@ -553,17 +554,21 @@ def extract_file(self, relative_path='', extract_path=None):
             if relative_path:
                 command.append(relative_path)
             LOGGER.info('Extracting file with: %s to %s', command, output_path)
-            rc = subprocess.call(command)
-            LOGGER.debug('Extract file RC: %s', rc)
-            if rc:
+            rc = subprocess.check_output(command)
+            if 'No files extracted' in rc:
                 raise StorageException(_('Extraction error'))
         else:
-            LOGGER.info('Copying AIP from: %s to %s', full_path, output_path)
-            shutil.copytree(full_path, output_path)
+            if relative_path:
+                # copy only one file out of aip
+                head, tail = os.path.split(full_path)
+                src = os.path.join(head, relative_path)
+                os.mkdir(os.path.join(extract_path, basename))
+                shutil.copy(src, output_path)
+            else:
+                src = full_path
+                shutil.copytree(full_path, output_path)
 
-        if not relative_path:
-            self.local_path_location = ss_internal
-            self.local_path = output_path
```

Comment:
@jhsimpson (@sevein @helenst) the deletion of these three lines seems to have introduced a bug, which was erroneously identified with [Redmine 11419](https://projects.artefactual.com/issues/11419#note-5). Because `self.local_path_location` is no longer being set after extraction, all re-ingest is failing. The chain of events is quite convoluted, but essentially all of the paths in `reingest_files` in `start_reingest()` end up being wonky, i.e., they end up looking like the impossible `var/archivematica/sharedDirectory/www/AIPsStore/var/archivematica/storage_service/...`

Long story short, I'm wondering if there is a reason that lines 564-566 were removed. If not, I'd like to revert this part of this PR.

Reply:
See https://github.com/artefactual/archivematica-storage-service/pull/235